src/pwm.rs

```diff
@@ -446,10 +446,15 @@ macro_rules! pwm_timer_private {
             }
 
             // enable auto reload preloader
-            tim.cr1.write(|w| w.arpe().set_bit());
+            tim.cr1.modify(|_, w| w.arpe().set_bit());
 
             // Set the "resolution" of the duty cycle (ticks before restarting at 0)
             // Oddly this is unsafe for some timers and not others
+            //
+            // FIXME: I guess this is due to some Timers not allowing values >16bit.
+            // So this needs to be checked according to the Timer type.
+            //
+            // NOTE(write): uses all bits of this register.
```

Comment:
Actually i think this unsafe is there because not all timers are documented in
`stm32f3`. And when they are not,this registers are marked as unsafe, because no
one programmatically confirmed that writing to this register has no side effects.

Reply:
Unsafe is here, because of the reason I mentioned above. See for example [TIM17](https://stm32.agg.io/rs/STM32F303.html#TIM17:ARR) which ARR registers are still not documented in stm32f3. 

However, justifying write does still make sense. Remove the FIXME and this PR is ready to be merged :)